File: 2015/Day1.cs
/*
 * Advent of Code 2015 day 1 solution in C#
 * http://adventofcode.com/2015/day/1
 * Lloyd Dilley
 */

using System;
using System.IO;

public class Day1
{
  public const char Up = '(';             // instruction to go up a floor
  public const char Down = ')';           // instruction to go down a floor
  public const string FileName = "1.dat"; // data file name

  public static void Main()
  {
    int position = 0;                     // current floor number
    string data = "";                     // data from input file

    try
    {
      data = File.ReadAllText(FileName);
    }
    catch(Exception e)
    {
      Console.WriteLine("Unable to parse input file:\n{0}", e);
      Environment.Exit(1);
    }

    for(int i = 0; i < data.Length; i++)
    {
      if(data[i] == Up)
        position++;
      if(data[i] == Down)
        position--;
      Console.WriteLine("Current floor: #{0}", position);
    }
    Console.WriteLine("Arrived at destination floor: #{0}", position);
    return;
  }
}

File: 2015/Day1.scala
/*
 * Advent of Code 2015 day 1 solution in Scala
 * http://adventofcode.com/2015/day/1
 * Lloyd Dilley
 */

import scala.io.Source

 object Day1
 {
  val UP = '('            // instruction to go up a floor
  val DOWN = ')'          // instruction to go down a floor
  val FILE_NAME = "1.dat" // data file name

  def main(args: Array[String])
  {
    try
    {
      var position = 0    // current floor number
      val data_file = Source.fromFile(FILE_NAME)
      val data = data_file.getLines.mkString
      data_file.close
      for(i <- 0 until data.length())
      {
        if(data.charAt(i) == UP)
        {
          position += 1
        }
        if(data.charAt(i) == DOWN)
        {
          position -= 1
        }
        println("Current floor: #" + position)
      }
      println("Arrived at destination floor: #" + position)
    }
    catch
    {
      case e: Exception =>
      {
        println("Unable to parse in put file: " + e)
      }
    }
  }
}